src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -4090,9 +4115,12 @@ WorkerLimitCount(MultiExtendedOp *originalOpNode)
 	List *groupClauseList = originalOpNode->groupClauseList;
 	List *sortClauseList = originalOpNode->sortClauseList;
 	List *targetList = originalOpNode->targetList;
+	bool orderByNonAggregates = !(HasOrderByAggregate(sortClauseList, targetList));
```

Comment:
You have this line below
(sortClauseList != NIL && orderByNonAggregates)

but still orderByNonAggregates has a wrong meaning in this case.